### PR TITLE
implement subscription registration with unit test

### PR DESCRIPTION
**Pull request by ankitdas13**
for create : https://github.com/razorpay/razorpay-node/blob/master/documents/card.md#create-registration-link
Link : https://razorpay.com/docs/api/recurring-payments/cards/authorization-transaction/#121-create-a-registration-link

Doc is in this PR : https://github.com/razorpay/razorpay-ruby/blob/update_readme/documents/registerEmandate.md#create-registration-link

### CODE REVIEW

File: lib/razorpay/subscription_registration.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require 'razorpay/request'
+require 'razorpay/entity'
+
+module Razorpay
+  # Addon API allows you to fetch and delete
+  # subscription-addons with Razorpay
+  class SubscriptionRegistration < Entity
```
Comment (sonyy-m) [inline lib/razorpay/subscription_registration.rb:7]: update class-level comments

File: test/fixtures/fake_subscription_registration.json [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        "name": "Gaurav Kumar",
+        "account_number": "11214311215411",
+        "account_type": "savings",
+        "beneficiary_email": "gaurav.kumar@example.com",
+        "beneficiary_mobile": "9123456780"
+      },
+      "first_payment_amount": 0
+    },
+    "nach_form_url": "https://rzp.io/i/jlERKqU"
+  }
```
Comment (sonyy-m) [inline test/fixtures/fake_subscription_registration.json:91]: fix indentation and add new line at end of file

File: test/razorpay/test_subscription_registration.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
+require 'test_helper'
+
+module Razorpay
+  # Tests for Razorpay::Addon
+  class RazorpayAddonTest < Minitest::Test
+    class SubscriptionRegistration < Razorpay::Entity; end
```
Comment (sonyy-m) [inline test/razorpay/test_subscription_registration.rb:6]: why is the `RazorpayAddonTest ` class used here?

File: lib/razorpay/subscription_registration.rb (hunk 1 above)
Comment (ankitdas13) [reply lib/razorpay/subscription_registration.rb]: have changed the comments

File: test/fixtures/fake_subscription_registration.json (hunk 2 above)
Comment (ankitdas13) [reply test/fixtures/fake_subscription_registration.json]: eol fixed

File: test/razorpay/test_subscription_registration.rb (hunk 3 above)
Comment (ankitdas13) [reply test/razorpay/test_subscription_registration.rb]: fixed class name

**Pull request merged**